extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -141,185 +76,217 @@ export abstract class AzureAuth implements vscode.Disposable {
 		this.scopesString = this.scopes.join(' ');
 	}
 
-	public abstract async login(): Promise<AzureAccount | azdata.PromptFailedResult>;
-
-	public abstract async autoOAuthCancelled(): Promise<void>;
-
-	public abstract async promptForConsent(resourceId: string, tenant: string): Promise<{ tokenRefreshResponse: TokenRefreshResponse, authCompleteDeferred: Deferred<void> } | undefined>;
-
-	public dispose() { }
-
-	public async refreshAccess(oldAccount: azdata.Account): Promise<azdata.Account> {
-		const response = await this.getCachedToken(oldAccount.key);
-		if (!response) {
-			oldAccount.isStale = true;
-			return oldAccount;
-		}
-
-		const refreshToken = response.refreshToken;
-		if (!refreshToken || !refreshToken.key) {
-			oldAccount.isStale = true;
-			return oldAccount;
+	public async startLogin(): Promise<AzureAccount> {
```

Comment:
`Promise<AzureAccount | undefined>`

Reply:
I should turn on strict null mode for azurecore, would help so much.